src/main/java/ga/rugal/food/core/dao/impl/MenuDaoImpl.java

```diff
@@ -70,5 +72,29 @@ public Menu deleteById(Integer id)
     {
         return Menu.class;
     }
+    
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    @Transactional(readOnly = true)
+    public int countMenusByRestaurant(Restaurant r) {     
+   
+        int count = countByProperty("restaurant", r);        
+        return count;
+        
+    }
+    
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    @Transactional(readOnly = true)
+    public Menu getRandomMenuByRestaurant(Restaurant r) {
+        Random rd = new Random();
```

Comment:
But why not to have a field member with `Random` object, just like what you have done in `RestaurantAction`